traitsui/testing/tester/tests/test_ui_wrapper.py

```diff
@@ -463,3 +463,43 @@ def handler(wrapper, action):
 
         with self.assertRaises(ZeroDivisionError):
             wrapper.perform(None)
+
+    def test_perform_event_processed_optional(self):
+        # Allow event processing to be switched off.
+        gui = GUI()
+        side_effect = mock.Mock()
+
+        def handler(wrapper, action):
+            gui.invoke_later(side_effect)
+
+        wrapper = example_ui_wrapper(
+            registries=[StubRegistry(handler=handler)],
+            process_events=False,
+        )
+
+        # With process_events set to False, events are not automatically
+        # processed.
+        wrapper.perform(None)
+
+        self.assertEqual(side_effect.call_count, 0)
```

Comment:
In the test for UITester, you say
```
# We still want the events (if any) to be processed at the end of
# the test.
self.addCleanup(process_cascade_events)
```
Is that not he case here as well, because when `wrapper.perform(None)` is called, `process_cascade_events` won't be, but `handler` still will be.

Reply:
You are right that I should not leave the event behind at the end of the test. I was being lazy here because the `side_effect` here is a local mock object that won't interact with other tests so it does not matter if it is called later, though I did leave the object on the stack for longer than it has to be. For the other test on `create_ui`, GUI widgets are created and then scheduled for disposal so if I don't make sure events are processed at the end of the tests, those GUI widgets could interact with the next test (through stealing focus / overlapping windows / etc...).

Reply:
Done adding cleanups for processing events.